lint/quick_fix.py

```diff
@@ -0,0 +1,597 @@
+):
+    # type: (str, str, str, Set[str]) -> Callable[[Fixer], Fixer]
+    def register(fn):
+        # type: (Fixer) -> Fixer
+        ns_name = namespacy_name(fn)
+        provider = partial(std_provider, subject, detail, except_for, fn)
+        PROVIDERS[linter_name][ns_name] = provider
+        fn.unregister = lambda: PROVIDERS[linter_name].pop(ns_name, None)  # type: ignore[attr-defined]
+        return fn
+
+    return register
+
+
+def namespacy_name(fn):
+    # type: (Callable) -> str
+    # TBC: No methods supported, only simple functions!
+    return "{}.{}".format(fn.__module__, fn.__name__)
+
+
+def std_provider(subject, detail, except_for, fixer, errors, _view):
+    # type: (str, str, Set[str], Fixer, List[LintError], Optional[sublime.View]) -> Iterator[QuickAction]
+    make_action = lambda error: QuickAction(
+        subject.format(**error),
+        partial(fixer, error),
+        detail.format(**error),
+        solves=[error]
+    )
+
+    grouped_by_code = group_by(
+        lambda e: e["code"],
+        (e for e in errors if e["code"] and e["code"] not in except_for)
+    )
+    for code, errors_with_same_code in grouped_by_code.items():
+        grouped_by_line = group_by(lambda e: e["line"], errors_with_same_code)
+
+        actions_per_line = []
+        for line, errors_on_same_line_with_same_code in grouped_by_line.items():
+            as_actions = list(map(make_action, errors_on_same_line_with_same_code))
+            actions_per_line.append(as_actions)
+
+        if len(actions_per_line) > 1:
+            yield merge_actions(actions_per_line)
+        else:
+            actions = actions_per_line[0]
+            head = actions[0]
+            if len(actions) == 1:
+                yield head
+            else:
+                yield QuickAction(
+                    subject_for_multiple_actions(actions),
+                    head.fn,
+                    detail_for_multiple_actions(actions),
+                    solves=list(flatten(action.solves for action in actions))
+                )
+
+
+def fix(linter_name, only_for=set()):
+    # type: (str, Union[str, Set[str]]) -> Callable[[Fixer], Fixer]
+    if isinstance(only_for, str):
+        only_for = {only_for}
+
+    def register(fn):
+        # type: (Fixer) -> Fixer
+        ns_name = namespacy_name(fn)
+        provider = partial(std_fix_provider, linter_name, only_for, fn)
+        PROVIDERS[linter_name][ns_name] = provider
+        fn.unregister = lambda: PROVIDERS[linter_name].pop(ns_name, None)  # type: ignore[attr-defined]
+        return fn
+    return register
+
+
+def std_fix_provider(linter_name, only_for, fixer, errors, _view):
+    return (
+        QuickAction(
+            "{linter}: Fix {code} {msg}".format(**error),
+            partial(fixer, error),
+            "",
+            solves=[error]
+        )
+        for error in errors
+        if error["code"] in only_for or not only_for
+    )
+
+
+# import zoo, foo
+# import bar
```

Comment:
🤔

Reply:
LOL, you didn't see all the fixups for the ws errors I initially committed. You need some test area.